cordapp-example/B9LAB-NOTES.md

```diff
@@ -0,0 +1,32 @@
+
+# B9lab Corda course notes
+
+## Module 3
+
+### Expand the example
+
+> To find out
+> - How do you pass a known state from the vault when launching a flow?
+
+I can use simply tx id, and in a code pull the state like so:
+`val ourStateRef = StateRef(SecureHash.parse(txId), 0)`
+
+> - What notary do you choose?
+
+I will use default notary.  
+But I'm also wondering if there is a way not to use it?
```

Comment:
Imagine this, the `borrower` says: this IOU is cancelled/destroyed because I paid it in full, but the `lender` will say: no it's not; where is my signature on the transaction attesting to that?  
  
You need to have a flow with 2 parts (Initiator and Responder), where the `borrower` starts the initiator and collects the signature of the `lender` (which runs the responder), to mimic the payment; you can use a `send` in the initiator and a `receive` in the resopnder, the `lender` will validate that the sent amount matches the IOU amount, and only then sign the transaction.

Link about `send` and `receive`: https://docs.corda.net/api-flows.html#receive